### ISSUE TITLE
Overwatch; Fog of war is broken in certain cases

### ISSUE THREAD

**Issue by Mabadman** [ValveSoftware/Dota2-Gameplay#6646]
### Description
Certain conditions will cause Overwatch replay to be bugged. Camera will be locked in place until either; a marker is reached, or the viewer skips to a different point in time using the seek bar. 
After this, the map will remain bugged (no trees/buildings, fog of war covers entire map, minimap broken) but units will be visible and camera will be freed.

It seems this is caused by the replay starting too early. In the example I'll provide, the replay begins as early as possible, as you can see below.
![image](https://user-images.githubusercontent.com/42493726/212542080-ad09abf5-e888-42d6-b3ee-4658186a3c9a.png)

### Example
Match ID for the Overwatch case: 6971235006
The camera will be locked in place (first half of the gif). This lasts for 10 minutes (at 1:1 speed) or until the replay watcher uses the seek bar to skip in time (going backwards works too).
Once the camera is freed and hud & units become visible, the map will remain bugged (second half of the gif)
![bug](https://user-images.githubusercontent.com/42493726/212542828-16ab4b7e-7c7d-4d18-bcdb-4468b299b423.gif)

**Comment by shaileshgarg** [comment]
![20230115235017_1](https://user-images.githubusercontent.com/26450759/212559616-e462e4b0-1695-494b-8586-e1e37bbc0bb6.jpg)

Encountered same issue. No camera movement first and map is dark.

In my case the first marker is at 15 min,, but still the overwatch starts from 0 min in the game. This didn't used to happen.

**Comment by anko** [comment]
I've only seen broken overwatch replays in matches where the reported player was clearly scripting. I suspect that their script is intentionally breaking the replay. In [match 6971941488](https://www.opendota.com/matches/6971941488), Silencer is scripting, and the game looks like this:

![shot](https://user-images.githubusercontent.com/5231746/212712216-5c2e17e1-08ae-4de9-8fba-641ff485b874.png)

Everything is fog, the camera zoom is odd, trees and buildings are not shown, the minimap is completely broken, and the _Prev. Marker_ and _Next Marker_ buttons don't work.

**Comment by shaileshgarg** [comment]
Interesting observation, in my case also it was a scripting issue which was clear ffrom the mouse movement.

**Comment by Mabadman** [comment]
It's really weird how it takes +10 minutes for the markers to start in both my case and anko's, and yet Overwatch includes the entirety of pregame in its "relevant clip".
Considering these are all cases of possible hacking/scripting, it seems very likely there's an exploit which allows these scripters to screw up Overwatch.
Thankfully it's not perfect so we can usually still deliver resolutions. But still, this should be looked into.

**Comment by FuckCheatersDota** [comment]
same thing has happened to me twice both times they were scripting, i think their scripts are messing with overwatch.

**Comment by TheMatheusDev** [comment]
I experienced this yesterday when overwatching an Arc Warden reported by scripting!

**Comment by zoonzoon2** [comment]
Yup, I got same bug twice. Both times in cases where suspect was cheating.

**Comment by zeroedout** [comment]
I've seen this on at least 4 different replays. If I recall correctly, all of them were suspected scripting cheaters. All of them had suspicious mouse movements.

**Comment by asmodeus941** [comment]
This is not a bug. This is a script that is breaking replays. This only happens when the person is reported for scripting and is actually scripting. Just report them for scripting.

**Comment by shaileshgarg** [comment]
> This is not a bug. This is a script that is breaking replays. This only happens when the person is reported for scripting and is actually scripting. Just report them for scripting.

That is the bug, any script the player is using shouldn't break overwatch. That is what overwatch is meant for.

**Comment by ShinigamiGamingInc** [comment]
Had it on different accessions, with different cheats being used. Also the bug changed over time, in the beginning it was just starting on hero spawn. With vision and camera working but trees where missing, now it the vision is bugged for me.
But i didn't notice the camera being stuck, for me it always jumped to the hero after starting of in the center.
Another thing i found was that the "move to next marker" dose not work until you get to the actual time frame of the case. (1min or so, bevor the first report)

**Comment by ShinigamiGamingInc** [comment]
> It's really weird how it takes +10 minutes for the markers to start in both my case and anko's, and yet Overwatch includes the entirety of pregame in its "relevant clip".

It's not, marker skipping dose not work in this time frame, only if you get in the right time frame for the case, so the system knows the right start point, but dose not restrict the beginning correctly.

**Comment by anko** [comment]
I think this is fixed. I done a lot of reviews, and have not seen a single glitched Overwatch replay since the [Banhammer 40k update](https://www.dota2.com/newsentry/3677788723152833273).

Anyone else still seeing this?

**Comment by shaileshgarg** [comment]
I also haven't encountered this after the bans
